File: examples/python/01_basic_flight.py
from easylink import Drone

def main():
    # 1. Instantiate & auto-connect (supports SITL or hardware serial)
    print("Initiating EASYLink connection...")
    drone = Drone()
    drone.connect()
    
    # 2. Pre-flight health check
    if drone.preflight_check():
        print("Pre-flight safety checks PASSED!")
    else:
        print("Pre-flight checks FAILED!")
        return

    # 3. Flight sequence
    print("Arming drone...")
    drone.arm()

    print("Taking off to 10.0m...")
    drone.takeoff(10.0)

    # 4. Read telemetry
    pos = drone.position
    att = drone.attitude
    bat = drone.battery
    print(f"Telemetry -> Pos: Lat={pos.lat}, Lon={pos.lon}, Alt={pos.alt}m")
    print(f"Telemetry -> Attitude: Roll={att.roll:.1f}°, Pitch={att.pitch:.1f}°, Yaw={att.yaw:.1f}°")
    print(f"Telemetry -> Battery: Voltage={bat.voltage}V ({bat.remaining_pct}%)")

    # 5. Hover and Return to Launch (blocking)
    print("Hovering for 5 seconds...")
    drone.hover(5.0)

    print("Returning home and landing...")
    drone.home(blocking=True)

    drone.disconnect()
    print("Flight sequence complete!")

if __name__ == "__main__":
    main()

File: examples/cpp/01_basic_flight.cpp
#include <easylink/drone.hpp>
#include <iostream>

int main() {
    std::cout << "Connecting to drone via EASYLink-CPP..." << std::endl;
    easylink::Drone drone("udp:127.0.0.1:14550");
    
    if (!drone.connect()) {
        std::cerr << "Failed to establish connection!" << std::endl;
        return 1;
    }

    std::cout << "Arming vehicle..." << std::endl;
    drone.arm();

    std::cout << "Taking off to 10m..." << std::endl;
    drone.takeoff(10.0f);

    auto pos = drone.position();
    std::cout << "Position: Lat=" << pos.lat << ", Lon=" << pos.lon << ", Alt=" << pos.alt << "m" << std::endl;

    std::cout << "Hovering for 5 seconds..." << std::endl;
    drone.hover(5.0f);

    std::cout << "Returning home..." << std::endl;
    drone.home(true);

    drone.disconnect();
    std::cout << "Flight complete!" << std::endl;
    return 0;
}
